modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Fields.tsx

```diff
@@ -0,0 +1,493 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayButton, {ClayButtonWithIcon} from '@clayui/button';
+import {ClayCheckbox, ClayInput} from '@clayui/form';
+import ClayLayout from '@clayui/layout';
+import ClayLoadingIndicator from '@clayui/loading-indicator';
+import ClayModal from '@clayui/modal';
+import {ManagementToolbar} from 'frontend-js-components-web';
+import {fetch, navigate, openModal, openToast} from 'frontend-js-web';
+import React, {useEffect, useRef, useState} from 'react';
+
+import {API_URL, OBJECT_RELATIONSHIP} from '../Constants';
+import {FDSViewSectionInterface} from '../FDSView';
+import {FDSViewType} from '../FDSViews';
+import OrderableTable from '../components/OrderableTable';
+
+type FDSFieldType = {
+	label: string;
+	name: string;
+	renderer: string;
+	sortable: boolean;
+	type: string;
+};
+
+type FieldType = {
+	name: string;
+	selected: boolean;
+	type: string;
+	visible: boolean;
+};
+
+interface AddFDSFieldsModalContentInterface {
+	closeModal: Function;
+	fdsView: FDSViewType;
+	onSave: Function;
+}
+
+const AddFDSFieldsModalContent = ({
+	closeModal,
+	fdsView,
+	onSave,
+}: AddFDSFieldsModalContentInterface) => {
+	const [fields, setFields] = useState<Array<FieldType> | null>(null);
+	const [query, setQuery] = useState('');
+
+	const onSearch = (query: string) => {
+		setQuery(query);
+
+		if (!fields) {
+			return;
+		}
+
+		const regexp = new RegExp(query, 'i');
+
+		setFields(
+			fields.map((field) => ({
+				...field,
+				visible: Boolean(field.name.match(regexp)),
+			}))
+		);
+	};
+
+	useEffect(() => {
+		const getFields = async () => {
```

Comment:
Why is this async? We don't call `await` when we call `getFields`. I know you are awaiting in the body of this function, but doesn't this still require an await when you call it?